controllers/component_pipelinerun_nudge_controller.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2021-2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v12 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/strings/slices"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+
+	"github.com/go-logr/logr"
+	applicationapiv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	contextTimeout = 300 * time.Second
+	// PipelineRunTypeLabel contains the type of the PipelineRunTypeLabel.
+	PipelineRunTypeLabel = "pipelines.appstudio.openshift.io/type"
+	// PipelineRunBuildType is the type denoting a build PipelineRun.
+	PipelineRunBuildType         = "build"
+	NudgeProcessedAnnotationName = "build.appstudio.openshift.io/component-nudge-processed"
+
+	// PipelineRunComponentLabel is the label denoting the application.
+	PipelineRunComponentLabel = "appstudio.openshift.io/component"
```

Comment:
nitpick: other controller and annotation above use `Name` suffix to distinguish label/key etc name from the value constant.